tests/JustSaying.Extensions.DependencyInjection.AwsCore.Tests/ClientFactoryTests.cs

```diff
@@ -31,76 +41,88 @@ public void ShouldReadConfigFromFile()
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
         var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
         var snsClient = clientFactory.GetSnsClient(RegionEndpoint.USEast1);
-        Assert.True(snsClient.Config.ServiceURL.Equals("http://profile.test/") == true);
-    }
 
+        //Assert
+        Assert.True(snsClient.Config.RegionEndpoint == null);
+        Assert.True(snsClient.Config.ServiceURL.Equals("http://profile.test/") == true, snsClient.Config.ServiceURL);
+    }
 
     [Fact]
     public void CanSetCorrectRegion()
     {
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
-        Environment.SetEnvironmentVariable("AWS_PROFILE", "local");
+        //Arrange
         var config = new ConfigurationBuilder()
             .AddEnvironmentVariables()
             .Build();
         var serviceCollection = new ServiceCollection();
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
+        var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
+        var sqsClient = clientFactory.GetSqsClient(RegionEndpoint.EUWest2);
+        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest2);
 
+        //Assert
+        Assert.True(sqsClient.Config.RegionEndpoint == RegionEndpoint.EUWest2);
+        Assert.True(snsClient.Config.RegionEndpoint == RegionEndpoint.EUWest2);
     }
 
     [Fact]
     public void ThrowsExceptionIfRegionConfigDiffers()
     {
-        Environment.SetEnvironmentVariable("AWS_PROFILE", null);
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
+        //Arrange
         var config = new ConfigurationBuilder()
             .AddEnvironmentVariables()
             .Build();
         var serviceCollection = new ServiceCollection();
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
         var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
-        Assert.Throws<ArgumentException>(() => clientFactory.GetSnsClient(RegionEndpoint.USEast1));
+
+        //Assert
+        Assert.Throws<ArgumentException>(() => clientFactory.GetSnsClient(RegionEndpoint.USEast2));
```

Comment:
Exception for Shouldy here - keep xunit's Assert as it has specific overloads for validating these types of exceptions, but change to use the one that validates the argument name, so:

```suggestion
        Assert.Throws<ArgumentException>("region", () => clientFactory.GetSnsClient(RegionEndpoint.USEast2));
```